streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -210,9 +214,8 @@ public void handleAssignment(final Map<TaskId, Set<TopicPartition>> activeTasks,
                 tasksToRecycle.add(task);
             } else {
                 try {
-                    task.prepareCloseClean();
-                    final Map<TopicPartition, OffsetAndMetadata> committableOffsets = task
-                        .committableOffsetsAndMetadata();
+                    task.suspend();
```

Comment:
question: I cannot remember why we need to commit those still owned tasks during handle-assignment, is that necessary? Or is that just an optimization: since we are going to commit anyways, let's just commit everyone.

If that's the case, we can refresh the last-commit timestamp as well.

Reply:
Even in that case, in line 205 we could check `task.commitNeeded() && task.isActive` right?

Reply:
For eoa-beta, if we commit, we alway need to commit all tasks. And to not distinguish between non-eos/eos-alpha vs eos-beta, be decided to just commit all tasks for all cases.

And we don't own `StreamThread#lastCommitMs` so we cannot update it.

For L205: the outter if checks already if `task.isActive`

Reply:
@mjsax @guozhangwang why do we need to commit at all during `handleAssignment`? Shouldn't we have already committed all tasks that _need_ to be committed during `handleRevocation`? 

That's not exactly a bug, I'm just wondering if it's necessary?

Reply:
We may not call `handleRevocation` before calling `handleAssignment` so the task to close may not be in SUSPENDED state yet, and hence do close them we need to commit their states. For other tasks, they are not necessarily committing but I think the point was, that since we are going to send one `commit` request anyways so just commit for everyone --- note that flushing can indeed be skipped, which is what KAFKA-9450 covers